Add Premake language definition

glpetrikov opened the pull request:
Add Premake as a separate language (premake5.lua)

## Description

Premake is a quite popular Lua-based build configuration tool used extensively in the C++  community.

Currently premake5.lua files are classified as generic Lua, which significantly distorts language statistics for many repositories.

This PR adds Premake as a distinct language with its canonical filenames, official brand color, and a  heuristic.

**Example of a notable project using Premake:**
- [Hazel Engine](https://github.com/TheCherno/Hazel) (12.7k stars, 1.6k forks) - Game engine by The Cherno, widely used as a learning resource in the C++ community


Official links:

- Repository: https://github.com/premake/premake-core

- Website: https://premake.github.io/

Real World examples:
Source: https://github.com/TheCherno/glfw/blob/9bed794ab7c1b961aaca259403695bbd3870d3b3/premake5.lua
License: zlib/libpng (GLFW)

## Checklist:

- [x] **I am adding a new language.**

  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.

    - Search results for each extension:

      - premake5.lua: https://github.com/search?q=path%3A**%2Fpremake5.lua+-is%3Afork&type=code

  - [x] I have included a real-world usage sample for all extensions added in this PR:

      - Sample source(s):
      **premake5.lua:**
```lua
      workspace "MyWorkspace"
        configurations { "Debug", "Release" }
      
      project "MyProject"
        kind "ConsoleApp"
        language "C++"
        files { "**.h", "**.cpp" }
      
        filter { "configurations:Debug" }
          defines { "DEBUG" }
          symbols "On"
      
        filter { "configurations:Release" }
          defines { "NDEBUG" }
          optimize "On"
```
      Source: https://premake.github.io/docs/What-Is-Premake
      License: BSD-3-Clause (Premake documentation)


  - [x] I have added a color

    - Hex value: #e67e22

    - Rationale: one of Premake's Official Brand Colors, used in the logo

  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

    - Matches Premake-specific top-level keywords while avoiding false positives on regular Lua scripts

lildude reviewed (changes requested):
Thanks for your detailed supporting explanation for this which makes sense to me. 

Along with my inline comments, we need a real-world sample for every filename added. Please add a sample for each and update the PR template to state where they come from (not your fork of linguist 😉) and the licence of each sample.

glpetrikov commented:
Thank you @lildude!

I've made all the necessary edits:
- Removed the heuristics section (not needed for filename-based detection)
- Removed the interpreters section (premake files don't use shebangs)  
- Added real-world samples for both premake4.lua and premake5.lua with sources and licenses

Ready for review!

lildude reviewed (changes requested):
> I have included a real-world usage sample for all extensions added in this PR:

You need to actually include the files in the PR, not the PR template, as they're used to train the classifier. We need at least one per filename being added with the template updated to link to each source (not your fork of linguist) with the license for each stated. The samples also need to be real-world examples so sourcing from the docs generally isn't a good indicator of real-world usage.

You'll also need to update the grammars README so that it reports the source of the grammar. The command to run is shown when you run the tests so I've approved CI to run on this PR.

glpetrikov commented:
Hey @lildude, I think I've made all the required changes.  
I deleted premake4.lua because it's long outdated and hard to find a real example.  
I also added a real-world premake5.lua example with source and license, and updated vendor/grammars/README.md with a link to it.

glpetrikov committed:
Add Premake language definition

lildude commented:
You've not resolved the ordering test failure.

glpetrikov committed:
Merge branch 'main' into premake

glpetrikov commented:
Ready for review!

I've removed `premake4.lua` because it is outdated.
I’ve also added a real-world `premake5.lua` example with source code and a license, and updated `vendor/README.md` using a script.
Everything is now in alphabetical order.

glpetrikov committed:
Merge branch 'main' into premake
